File: Max-Min Value/min_max_value.py
import numpy as np


def minimum(arr):
    """
    Finds the minimum of an array and returns it
    :param arr: Input integer or floating point array
    :return: The minimum value
    Complexity: ~O(n), S(1)
    """

    # Initialize the maximum number possible
    min_value = 999999999

    # Traverse the whole array and check if there are smaller elements
    for element in arr:
        if element < min_value:
            min_value = element
    return min_value


def maximum(arr):
    """
    Finds the maximum of an array and returns it
    :param arr: Input integer or floating point array
    :return: The maximum value
    Complexity: ~O(n), S(1)
    """

    # Initialize the minimum number possible
    max_value = -999999999

    # Traverse the whole array and check if there are any bigger elements
    for element in arr:
        if element > max_value:
            max_value = element

    return max_value


def main():
    """ Driver function."""

    # Initializing the testing array
    test_arr = [32.4, 21.7, 174.9, 18.3, 46.2, 68.1, 98.6, 13.4, 23.5, 108.4]

    # Calling and printing the minimum() function
    print("The minimum value of {} array is {}.\n".format(test_arr, minimum(test_arr)))

    # Calling and printing the maximum() function
    print("The maximum value of {} array is {}.\n".format(test_arr, maximum(test_arr)))


if __name__ == '__main__':
    main()

File: Max-Min Value/min_max_value.cpp
#include <iostream>


int minimum(int arr[], int size){
  /* Finds the minimum value of an array and returns it.
  Complexity: ~O(n), S(1) */
  
  // Initializing the biggest number possible
  int min_value = 9999999;
  
  // Traverse the whole array and check if there are smaller elements
  for (int i=0; i<size; i++){
    if (arr[i] < min_value)
      min_value = arr[i];
  }

  return min_value;
}


int maximum(int arr[], int size){
  /* Finds the maximum value of an array and returns it.
  Complexity: ~O(n), S(1) */
  
  // Initializing the biggest number possible
  int max_value = -9999999;
  
  // Traverse the whole array and check if there are smaller elements
  for (int i=0; i<size; i++){
    if (arr[i] > max_value)
      max_value = arr[i];
  }

  return max_value;
}


int main() {
  
  int test_arr[] = {32, 21, 174, 18, 46, 98, 13, 23, 108};

  // Calculating the array size before calling to make it easier
  int size = sizeof(test_arr) / sizeof(test_arr[0]);
  
  std::cout << "The maximum value of the test array is: " << maximum(test_arr, size) << std::endl;
  
  std::cout << "The minimum value of the test array is: " << minimum(test_arr, size) << std::endl;

  return 0;
}
